deepspeed/runtime/engine.py

```diff
@@ -2673,25 +2685,35 @@ def get_sparse_tensor_module_names(original_set,
         return load_path, client_state
 
     def _load_zero_checkpoint(self, load_dir, tag, load_optimizer_states=True):
-        zero_sd_list = self._get_all_zero_checkpoints(load_dir, tag)
-        if zero_sd_list is None:
-            return False
-
-        if load_optimizer_states and self.dp_world_size != self.loaded_checkpoint_dp_world_size:
-            raise ZeRORuntimeException("The checkpoint being loaded used a DP " \
-                f"world size of {self.loaded_checkpoint_dp_world_size} but the " \
-                f"current world size is {self.dp_world_size}. Automatic adjustment " \
-                "of ZeRO's optimizer state partitioning with a new world size is not " \
-                "currently supported.")
+        if self.load_universal_checkpoint():
+            zero_sd_list = None
+            checkpoint_folder = f'{os.path.join(load_dir, tag)}'
+        else:
+            if load_optimizer_states and self.dp_world_size != self.loaded_checkpoint_dp_world_size:
+                raise ZeRORuntimeException("The checkpoint being loaded used a DP " \
+                    f"world size of {self.loaded_checkpoint_dp_world_size} but the " \
+                    f"current world size is {self.dp_world_size}. Automatic adjustment " \
+                    "of ZeRO's optimizer state partitioning with a new world size is not " \
+                    "currently supported.")
```

Comment:
Should we recommend here to use of the universal checkpoint tool in the error message?

Reply:
Absolutely, except there is no such tool in the deepspeed repo. We only have it inside Meg-DS at the moment.

Reply:
such a tool at the moment is very client- or should I say model-specific as it has to do with individual param names and prescribes how each should be merged.

I think most of it can be abstracted to the deepspeed side and have the client only provide a few dicts with merging rules, so the client side script would be very thin.

Reply:
eventually we also want to expand this functionality to all ZeRO stages, so that `zero_to_fp32.py` will become much much simpler than its current very difficult to follow version.